src/dataflow/internal/process/functions/function-call.ts

```diff
@@ -107,6 +110,32 @@ export function processFunctionCall<OtherInfo>(functionCall: RFunctionCall<Other
 		inIds.push(...functionName.in, ...functionName.unknownReferences)
 	}
 
+	// parse a source call and analyze the referenced code
+	if(named && functionCallName == 'source') {
+		const sourceFile = functionCall.arguments[0]
+		if(sourceFile?.value?.type == RType.String) {
+			const executor = new RShellExecutor()
+			const path = removeTokenMapQuotationMarks(sourceFile.lexeme)
+
+			// parse, normalize and dataflow the sourced file
+			const parsed = executeSingleSubStep('parse', {
+				request:                'file',
+				content:                path,
+				ensurePackageInstalled: true
```

Comment:
i mean technically we shouldn't be here if it isn't but better be safe than sorry i guess.